### PR TITLE
Add version check to IntelliJ Extension

### LINKED ISSUE #289: Add version check to IntelliJ extension
The VS Code extension and the CLI both check for a new version on activation and prompt the user to update if their version is outdated. This should be added to the IntelliJ extension as well.

### PR DESCRIPTION
**Pull request by infiniteregrets**
closes #289

### CODE REVIEW

File: intellij-ext/src/main/kotlin/com/metalbear/mirrord/SemverCheck.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import com.intellij.ide.plugins.PluginManagerCore
+import com.intellij.openapi.extensions.PluginId
+import java.net.URL
+import com.github.zafarkhaja.semver.Version;
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+
+class SemverCheck : ProjectManagerListener {
+    private val version: String? = PluginManagerCore.getPlugin(PluginId.getId("com.metalbear.mirrord"))?.version
+    private val versionCheckEndpoint: String = "https://version.mirrord.dev/get-latest-version?source=1&version=$version"
```
Comment (eyalb181) [inline intellij-ext/src/main/kotlin/com/metalbear/mirrord/SemverCheck.kt:15]: Source should be 3 (1 is VS Code, 2 is CLI)

File: intellij-ext/src/main/kotlin/com/metalbear/mirrord/DonotShowAction.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+//    constructor(text: String, description: String) {
+//        this(text, description)
+//    }
+//    override fun actionPerformed(e: AnActionEvent) {
+//        TODO("Not yet implemented")
+//    }
+//
+
+
+internal abstract class DonotShowAction(text: String, description: String, icon: Icon) :
```
Comment (eyalb181) [inline intellij-ext/src/main/kotlin/com/metalbear/mirrord/DonotShowAction.kt:17]: DoNot**

File: intellij-ext/src/main/kotlin/com/metalbear/mirrord/MirrordListener.kt [hunk 3]
```diff
@@ -103,7 +103,11 @@ class MirrordListener : ExecutionListener {
     }
 
     private fun getRunConfigEnv(env: ExecutionEnvironment): LinkedHashMap<String, String> {
-        val envMethod = env.runProfile.javaClass.getMethod("getEnvs")
+        val method = when (env.runProfile::class.simpleName) {
```
Comment (eyalb181) [inline intellij-ext/src/main/kotlin/com/metalbear/mirrord/MirrordListener.kt:106]: Is this related to the version check?

File: intellij-ext/src/main/kotlin/com/metalbear/mirrord/SemverCheck.kt [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    private fun checkVersion(project: Project) {
+        val remoteVersion = Version.valueOf(URL(versionCheckEndpoint).readText())
+//        val localVersion = Version.valueOf(version)
+        val localVersion = Version.valueOf("2.12.1")
+        if (localVersion.lessThan(remoteVersion)) {
+            // TODO: fix this to open the plugins window
+//            val action = DonotShowAction()
+            val notification = MirrordEnabler.notifier("New version of mirrord $remoteVersion is available! Update: Preferences->Plugins->mirrord", NotificationType.INFORMATION)
```
Comment (eyalb181) [inline intellij-ext/src/main/kotlin/com/metalbear/mirrord/SemverCheck.kt:32]:
```suggestion
            val notification = MirrordEnabler.notifier("New version of mirrord $remoteVersion is available! Update: Preferences -> Plugins -> mirrord", NotificationType.INFORMATION)
```

**Review by eyalb181 — CHANGES_REQUESTED** [review]

File: intellij-ext/src/main/kotlin/com/metalbear/mirrord/SemverCheck.kt [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                            setOf<@receiver:NotNull PluginId>(pluginId),
+                            true
+                        ) {
+                            notification.expire()
+                        }
+                    }
+                })
+                .addAction(object : NotificationAction("Don't show again") {
+                    override fun actionPerformed(e: AnActionEvent, notification: Notification) {
+                        semverCheckEnabled = false
```
Comment (eyalb181) [inline intellij-ext/src/main/kotlin/com/metalbear/mirrord/SemverCheck.kt:51]: Is this choice persisted somewhere? What happens if I choose "Don't show again" and then restart the IDE?

File: intellij-ext/src/main/kotlin/com/metalbear/mirrord/SemverCheck.kt [hunk 6]
```diff
@@ -0,0 +1,63 @@
+package com.metalbear.mirrord
```
Comment (eyalb181) [inline intellij-ext/src/main/kotlin/com/metalbear/mirrord/SemverCheck.kt:1]: Please rename file to VersionCheck.kt (semver is an implementation detail - also it should have been capitalized SemVer)

Comment (infiniteregrets) [reply intellij-ext/src/main/kotlin/com/metalbear/mirrord/SemverCheck.kt]: thanks, done

**Review by eyalb181 — APPROVED** [review]

**Comment by infiniteregrets** [comment]
merging as it works smoothly for me
